src/flint/FlintTypes.jl

```diff
@@ -6204,10 +6204,10 @@ end
 const fqPolyRepMatrixSpace = AbstractAlgebra.Generic.MatSpace{fqPolyRepFieldElem}
 
 mutable struct fqPolyRepMatrix <: MatElem{fqPolyRepFieldElem}
-  entries::Ptr{Nothing}
+  entries::Ptr{fqPolyRepFieldElem}
```

Comment:
I think this is not 100% true, since otherwise I would expect the following to work (before this PR, with htis PR the Ptr conversion becomes unnecessary):
```julia
julia> F = Native.GF(2,2);

julia> a = matrix(F, 2, 2, [1,2,3,4]);

julia> unsafe_load(Ptr{fqPolyRepFieldElem}(a.entries))

[1054580] signal (11.1): Segmentation fault

Reply:
It definitely is a lie, just like with many other `Ptr{ZZRingElem}` etc. etc. in our code base -- our Julia types are "supersets" of the FLINT types.

BTW you can get the same crash in `master` via `unsafe_load(Nemo.mat_entry_ptr(a, 1, 1))`. I simply chose the signatures here to make the implementation of `mat_entry_ptr` as simple as possible for each type.

We could of course also resolve this by adding a separate struct `fq_nmod_t` (and `fmpz_t` etc. etc.) which faithfully replicate the FLINT structs, nothing more and nothing less. Then e.g. `fqPolyRepFieldElem` could start with a member `x:: fq_nmod_t`. 

That way, we could resolve all the lies everywhere. But that would be a major piece of work for relatively limited gain (if I were to *start* Nemo today, that's more or less what I'd do anyway, but as it is now...).

Anyway: I am happy to discuss this further, but I think for the purpose of this PR, it doesn't really make anything worse, as nothing in Nemo.jl ever accesses `.entries` fields, and the access to `.rows` is limited to very, very few places. And to me at least it was helpful to get a better idea of what each `entries` and `rows` points to. But I am also ok with changing them back to `Nothing` and adjust the `mat_entry_ptr` to insert suitable `reinterpret` calls.

Reply:
Thanks for the explanation. I am fine with merging this. Could you open an issue with basically what you explained here, so that we keep it somewhere? (even though that's nothing to be changed in the near future (unless someone has way too much time))